File: portal/src/lib/auth/passwordPolicy.ts
/** Politique mot de passe ViroTeam (inscription + changement). */
export const PASSWORD_POLICY_HINT =
  "8 caractères minimum, avec une majuscule, une minuscule et un chiffre.";

const MIN_LENGTH = 8;
const HAS_UPPERCASE = /[A-ZÀ-ÖØ-Þ]/;
const HAS_LOWERCASE = /[a-zà-öø-ÿ]/;
const HAS_DIGIT = /[0-9]/;

/** Retourne `null` si valide, sinon un message d’erreur en français. */
export function validatePassword(password: string | null | undefined): string | null {
  const value = password ?? "";
  if (!value) return "Le mot de passe est requis.";
  if (value.length < MIN_LENGTH) return `Au moins ${MIN_LENGTH} caractères.`;
  if (!HAS_UPPERCASE.test(value)) return "Au moins une majuscule.";
  if (!HAS_LOWERCASE.test(value)) return "Au moins une minuscule.";
  if (!HAS_DIGIT.test(value)) return "Au moins un chiffre.";
  return null;
}

export function isPasswordValid(password: string): boolean {
  return validatePassword(password) === null;
}

File: lib/utils/password_policy.dart
import 'package:viro_team_v2/copy/app_copy.dart';

/// Politique mot de passe ViroTeam (inscription + changement).
abstract final class PasswordPolicy {
  static const int minLength = 8;

  /// Texte d’aide affiché sous les champs mot de passe.
  static String get hint => AppCopy.auth.passwordHint;

  /// Retourne `null` si valide, sinon un message d’erreur en français.
  static String? validate(String? password) {
    if (password == null || password.isEmpty) {
      return AppCopy.auth.passwordRequired;
    }
    if (password.length < minLength) {
      return AppCopy.auth.passwordMinLength(minLength);
    }
    if (!_hasUppercase.hasMatch(password)) {
      return AppCopy.auth.passwordNeedUpper;
    }
    if (!_hasLowercase.hasMatch(password)) {
      return AppCopy.auth.passwordNeedLower;
    }
    if (!_hasDigit.hasMatch(password)) {
      return AppCopy.auth.passwordNeedDigit;
    }
    return null;
  }

  static bool isValid(String password) => validate(password) == null;

  static final RegExp _hasUppercase = RegExp(r'[A-ZÀ-ÖØ-Þ]');
  static final RegExp _hasLowercase = RegExp(r'[a-zà-öø-ÿ]');
  static final RegExp _hasDigit = RegExp(r'[0-9]');
}
